update_citation.py

```diff
@@ -0,0 +1,38 @@
+'''Updates the CITATION.cff file:
+    - Sets the date-released to the current date
+    - Sets the version from toplevel package.json
+'''
+
+from datetime import datetime
+import json
+import re
+
+CITATION_FILE = 'CITATION.CFF'
+PACKAGE_FILE = 'package.json'
+VERSION_PATTERN = r'^version:\s+.*$'
+DATE_RELEASED_PATTERN = r'^date-released:.*$'
+VERSION = None
+TODAY = datetime.today().strftime('%Y-%m-%d')
+
+with open(PACKAGE_FILE, 'r') as package_file:
+    package_json = json.load(package_file)
+    try:
+        VERSION = package_json.get('version')
+    except KeyError:
+        print('No version found in root package.json')
+        VERSION = '0.0.0'
```

Comment:
Setting the version in the citation to 0.0.0 in this case doesn't look like something we would actually want. I would expect one of the following:

- raise an error
- remove the `version` key from CITATION.cff

Reply:
Yeah you're right. I think just raise an error to be honest. This is not a generalised solution; not finding a version is not supposed to happen.